### PR TITLE
RUBY-2695 Add Ruby versions to proxy testing

### PR DESCRIPTION
**Pull request by comandeo**

### CODE REVIEW

File: .evergreen/config-atlas.yml [hunk 1]
```diff
@@ -807,17 +807,25 @@ axes:
         display_name: No
 buildvariants:
   - matrix_name: "all"
-    matrix_spec: { ruby: '*' }
-    #matrix_spec: { }
+    matrix_spec:
+      # We exclude ruby 2.3 because we do not have this version on Ubuntu 18.04.
+      # Since it is planned to drop support of this version from the driver
+      # in the next release, it does not worth to invest time into supporting
```
Comment (p-mongo) [inline .evergreen/config-atlas.yml:813]: it is not worth it to invest time

**Review by p-mongo — APPROVED** [review]

**Comment by p-mongo** [comment]
No JRuby?

File: .evergreen/config-atlas.yml (hunk 1 above)
Comment (comandeo) [reply .evergreen/config-atlas.yml]: 👍

File: .evergreen/config/atlas.yml.erb [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
   - matrix_name: "serverless"
-    matrix_spec: { ruby: 'ruby-3.0' }
-    #matrix_spec: { }
+    matrix_spec:
+      # We exclude ruby 2.3 because we do not have this version on Ubuntu 18.04.
+      # Since it is planned to drop support of this version from the driver
+      # in the next release, it does not worth to invest time into supporting
+      # this version.
+      ruby: ["ruby-2.4", "ruby-2.5", "ruby-2.5", "ruby-2.6", "ruby-2.7", "ruby-3.0"]
```
[the remark this answers is not in the document]
Comment (comandeo) [reply .evergreen/config/atlas.yml.erb]: Thanks, fixed.

**Comment by comandeo** [comment]
> No JRuby?

Thanks, added.

**Pull request merged**